### PR TITLE
Suppress verbose "Sending build context" messages in non tty scenarios

### LINKED ISSUE #25661: Detect tty for high-fidelity "Sending build context" cli message
Client:
 Version:      1.12.0
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   8eab29e
 Built:        Thu Jul 28 22:00:36 2016
 OS/Arch:      linux/amd64

---

When starting a build, the "Sending build context" message doesn't seem to suppress spamming of stdout when it's going to a log file (our CI server console logs).

```
$ docker build -t myimage .
Sending build context to Docker daemon 557.1 kB
Sending build context to Docker daemon 1.114 MB
Sending build context to Docker daemon 1.671 MB
Sending build context to Docker daemon 2.228 MB
Sending build context to Docker daemon 2.785 MB
Sending build context to Docker daemon 3.342 MB
Sending build context to Docker daemon 3.899 MB
Sending build context to Docker daemon 4.456 MB
Sending build context to Docker daemon 5.014 MB
Sending build context to Docker daemon 5.571 MB
Sending build context to Docker daemon 6.128 MB
Sending build context to Docker daemon 6.685 MB
Sending build context to Docker daemon 7.242 MB
Sending build context to Docker daemon 7.799 MB
Sending build context to Docker daemon 8.356 MB
Sending build context to Docker daemon 8.913 MB
Sending build context to Docker daemon  9.47 MB
Sending build context to Docker daemon 10.03 MB
Sending build context to Docker daemon 10.58 MB
Sending build context to Docker daemon 11.14 MB
Sending build context to Docker daemon  11.7 MB
Sending build context to Docker daemon 12.26 MB
Sending build context to Docker daemon 12.81 MB
Sending build context to Docker daemon 13.37 MB
Sending build context to Docker daemon 13.93 MB
Sending build context to Docker daemon 14.48 MB
Sending build context to Docker daemon 15.04 MB
Sending build context to Docker daemon  15.6 MB
Sending build context to Docker daemon 16.15 MB
Sending build context to Docker daemon 16.71 MB
Sending build context to Docker daemon 17.27 MB
Sending build context to Docker daemon 17.83 MB
<---snip 3 pages of text--->
```

### PR DESCRIPTION
**Pull request by yongtang**
**\- What I did**

This fix tries to address the issue raised in #25661 where verbose "Sending build context" messages were not suppressed in non tty (e.g. log file) scenarios.

**\- How I did it**

This fix suppress the "Sending build context" message (other than the last one) if the output is not a tty. The last update of the "Sending build context" message is still sent to the output.

**\- How to verify it**

This fix has been tested manually:

```
docker build -t tmp . > build.log
```

The `build.log` is (only one "Sending build context"):

```
Sending build context to Docker daemon 249.4 MB
Step 1 : FROM busybox
 ---> 2b8fd9751c4c
Step 2 : RUN echo 'test'
 ---> Using cache
 ---> 7db298b6c493
Successfully built 7db298b6c493
```

**\- Description for the changelog**

**\- A picture of a cute animal (not mandatory but encouraged)**

This fix fixes #25661.

Signed-off-by: Yong Tang yong.tang.github@outlook.com

### CODE REVIEW

**Comment by tonistiigi** [comment]
LGTM

File: api/client/image/build.go [hunk 1]
```diff
@@ -211,6 +227,9 @@ func runBuild(dockerCli *client.DockerCli, options buildOptions) error {
 
 	// Setup an upload progress bar
 	progressOutput := streamformatter.NewStreamFormatter().NewProgressOutput(progBuff, true)
+	if !dockerCli.IsTerminalOut() {
```
Comment (tonistiigi) [inline api/client/image/build.go]: nit: maybe worth doing if/else in here to avoid unnecessary calls.

**Comment by vdemeester** [comment]
LGTM :cat: with the same nit as @tonistiigi

**Comment by yongtang** [comment]
Thanks @tonistiigi @vdemeester. The pull request has been updated.

File: api/client/image/build.go (hunk 1 above)
Comment (tonistiigi) [inline api/client/image/build.go]: Sorry, did I read it wrong? (can't see old diff anymore). I might have missed that `progressOutput` was reused.

Comment (yongtang) [inline api/client/image/build.go]: @tonistiigi `progressOutput` was reused as `lastProgressOutput` is just a wrapper on top. I think either way might be fine.

Comment (tonistiigi) [inline api/client/image/build.go]: My bad then. I think `&lastProgressOutput{progressOutput}` is better. Sorry for the back and forth.

Comment (yongtang) [inline api/client/image/build.go]: @tonistiigi No a problem. The pull request has been updated. And thanks for spending time to review the PR. 😃

**Comment by tonistiigi** [comment]
LGTM, ping @vdemeester

**Comment by vdemeester** [comment]
LGTM :cat:

**Pull request merged**